clusters/migrating-databases-to-gcp.md

```diff
@@ -0,0 +1,100 @@
+## Migrating databases to GCP postgreSQL
+
+Suggested patterns for moving on-prem databases to GCP postgreSQL.
+
+Disclaimer: These are options for migrations to GCP postgreSQL. Others may work better for your team.
+
+## Prerequisites
+The team needs to update their ROS and PVK analysis to migrate to GCP.
+Refer to the ROS and PVK section under [Google Cloud Platform clusters][GCP].
+
+See database creation in GCP in [Google Cloud Platform persistence][DBCREATION].
+
+## Migration paths available
+
+### From on-premise Oracle
+
+#### Replication migration using migration application
+
+Create a simple migration application that supports writing data to the new database. 
+Using requests sent to this application you can populate the new postgreSQL database.
+
+Rewrite the oracle DDL scripts to postgreSQL. If your oracle database contains specific oracle procedures or functions, 
+that do not exist in postgreSQL, they will have to be recreated in some other way.
+There are tools available to help ease this rewrite, for example ora2pg(http://ora2pg.darold.net/start.html). 
+Create the postgreSQL database in GCP and start deploy the application to GCP with the empty database 
+and let flyway (or other database versioning software) create the DDLs.
+
+Create migration app as a container in the same pod as the database application (this is to avoid permission issues using the same database). This migration application only handles the data transfer from the oracle database to postgreSQL in GCP. 
+Example: [PAM AD Migration][PAMADMIGRATION]
```

Comment:
Vi så på dette i sommer, og valgte å gjøre det med å mappe direkte til MapSqlParameterSource, i steden for å bruke JPA, som i prosjektet over. Fordelen var at det ble mindre kode, og brukte mye mindre minne, slik at vi også kunne gjøre migreringen(200 000+ rader) i en omgang, uten å bruke cursor. 

https://github.com/navikt/rekrutteringsbistand-kandidat-api-migrering